packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2734,6 +2725,25 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
+    private submitToBatch(batchManager: BatchManager, message: BatchMessage) {
+        const baseLine = batchManager.pendingBatchBaseline;
+        if (this.mc.config.getBoolean(disableBatchBaselineCheckKey) !== true
+            && baseLine !== undefined
+            && baseLine !== message.referenceSequenceNumber) {
+            const error = new GenericError(
```

Comment:
this check is weaker than direct check. I.e. if only first op is sent in response to processing op, and ref seq number changes at the beginning of op processing, then it will not capture violation (though you can argue - there is no violation in such case). I think the worse part - when it fires, it's not actionable, as we have no data to point to offender.
So some more direct check (essentially moving logic from DM to ContainerRuntime) would be probably better.

Reply:
> I think the worse part - when it fires, it's not actionable, as we have no data to point to offender.

We really can't at this layer. As mentioned in the ADO item, user experience it definitely lacking as we don't (and should not!) have a lot of information relevant to the user in this part of the code about where the op came from and why. This is a DDS related usage pattern and the DDS should handle it (or better yet the DDS base, which we own). I think we can deal with the inconsistency first, then improve the experience in SharedObject. This aims to ensure the invariant is not broken, the latter is to make it friendlier. I will separate the work items.

Reply:
We can collect stack and stack should be sufficient to identify the offender (assuming stack is collected when offense happens).